### PR TITLE
Remove specification files that have been moved to 'eligibility-api' project

### PR DESCRIPTION
**Pull request by angela-tran**
### What this PR does
cal-itp/eligibility-api#7 copied over and made edits to the `docs/specification` files from this project. This PR removes those files from the `eligibility-server` docs.

It also updates the URLs in the README.md file (even though the API docs aren't publishing yet).

### Tested locally
The `Specification` link is no longer present.

image

### CODE REVIEW

File: README.md [hunk 1]
```diff
@@ -1,9 +1,9 @@
 # Eligibility Server
 
-Server implementation of the [Eligibility Verification API](https://docs.calitp.org/benefits/eligibility-verification/). See
+Server implementation of the [Eligibility Verification API](https://docs.calitp.org/eligibility-api/specification). See
 the client implementation in [`benefits`](https://github.com/cal-itp/benefits).
 
-View the technical documentation online: [https://docs.calitp.org/eligibility-server](https://docs.calitip.org/eligibility-server)
+View the technical documentation online: [https://docs.calitp.org/eligibility-server](https://docs.calitp.org/eligibility-server)
```
Comment (machikoyasuda) [inline README.md:6]: great catch! ty!

**Review by machikoyasuda — APPROVED** [review]

**Comment by thekaveman** [comment]
Note this should not be merged until the API docs are published online! (Still need a GitHub Actions workflow)

**Comment by angela-tran** [comment]
> Note this should not be merged until the API docs are published online! (Still need a GitHub Actions workflow)

Good point - I will set this as a Draft until then. Thanks!

**Comment by angela-tran** [comment]
API docs are published online! (thanks to combo of cal-itp/eligibility-api#8 and cal-itp/eligibility-api#11)

Check it out at: https://docs.calitp.org/eligibility-api/

image

**Comment by machikoyasuda** [comment]
@angela-tran Looks great!

**Pull request merged**